Chunks/ChunkManagement/ChunkWorkers/ChunkWorker.cs

```diff
@@ -169,13 +171,35 @@ private void SaveChunks()
 
     private void UpdateChunks()
     {
-        if (!_chunksToUpdateQueue.TryDequeue(out var chunk))
+        if (!_modificationsToPerform.TryDequeue(out var modification))
         {
             IsUpdating = false; // TODO is this ever hit?
             return;
         }
 
-        chunk.Mesh.Vertices = _meshGenerator.GetMesh(chunk.Position, new ChunkData { SphereId = 0, Voxels = chunk.Voxels });
+        var modificationType = modification.ModificationType;
+        ref var location = ref modification.Location;
+        var time = modification.Time;
+        var brushWeight = modification.BrushWeight;
+        var radius = modification.Radius;
+        var chunk = modification.Chunk;
+
+        if (modificationType == ModificationType.Mine)
+        {
+            chunk.Mine(location, time, brushWeight, radius);
+        }
+        else if (modificationType == ModificationType.Build)
+        {
+            chunk.Build(location, time, brushWeight, radius);
+        }
+        else
+            throw new NotImplementedException();
+
+        var mesh = _meshGenerator.GetMesh(chunk.Position, new ChunkData { SphereId = 0, Voxels = chunk.Voxels });
+
+        lock (_lockUpdatedChunk)
```

Comment:
do we want a single lock for all chunks?

Reply:
No that's a mistake, I'll fix that